### PR TITLE
refactor combine.ts

### PR DESCRIPTION
**Pull request by Soolan**
- replace lodash methods with native js
- simplify the code
- add a new method for transaction validation

### CODE REVIEW

**Comment by intelliot** [comment]
This will fix https://github.com/ripple/ripple-lib/issues/987

File: src/transaction/combine.ts [hunk 1]
```diff
@@ -1,39 +1,27 @@
...
-}
-
-function compareSigners(a, b) {
-  return addressToBigNumber(a.Signer.Account)
-    .comparedTo(addressToBigNumber(b.Signer.Account))
+function hashArrayToObjectArray( signedTransactions : Array<string> ): Array {
+    validate.combine({signedTransactions})
+    const transactions: any[] = signedTransactions.map(binary.decode)
+    const tx = transactions[0]
+    delete tx.Signers
```
Comment (FKSRipple) [inline src/transaction/combine.ts]: you can remove this line, since you delete tx.Signers a few lines later

File: src/transaction/combine.ts [hunk 2]
```diff
@@ -1,39 +1,27 @@
...
+    transactions.forEach( _tx => {
+        delete _tx.Signers;
```
Comment (FKSRipple) [inline src/transaction/combine.ts]: can you add a comment here for why you're doing this?

File: src/transaction/combine.ts [hunk 3]
```diff
@@ -1,39 +1,27 @@
...
-      'txJSON is not the same for all signedTransactions')
-  }
-  const unsortedSigners = _.reduce(txs, (accumulator, _tx) =>
-    accumulator.concat(_tx.Signers || []), [])
-  const signers = unsortedSigners.sort(compareSigners)
-  const signedTx = _.assign({}, tx, {Signers: signers})
-  const signedTransaction = binary.encode(signedTx)
-  const id = computeBinaryTransactionHash(signedTransaction)
-  return {signedTransaction, id}
+    const txs = hashArrayToObjectArray(signedTransactions)
```
Comment (FKSRipple) [inline src/transaction/combine.ts]: nit: seems like the indentation is off

**Comment by Soolan** [comment]
> @Soolan thanks for the PR, I left a few comments. I know it's been a while, but are you still interested in getting this merged? Let me know and I'll help get it merged

Hi @FKSRipple 
Thank you for your response.
By all means please help me to merge that.


p.s.
I did study/review the code a while ago, and contributed tiny bits here and there, hoping to get the job Ripple advertised a while back. 
I would LOVE to work on this repository on a more serious level. (contract, permanent, anything! as long as I am contributing to Ripple I am happy).

If your offer still stands, please give me 8 days to deliver my current contract and then lets talk about it.

Thank you.

In the mean time, I wonder if you are kind enough to take care of the changes please. I agree with all your comments.

**Comment by FKSRipple** [comment]
@Soolan if you grant me access to edit the files on your branch, I'll make the recommended changes.

/cc @intelliot re: contracting

**Comment by Soolan** [comment]
So sorry @FKSRipple 
Last month I did a big cleanup on all my repos and deleted my ripple-lib fork as well.
I can't restore it because github keeps only (non-forked) deleted repos (for 90 days).

I've deleted my local files as well.
Give me a few days to finish my current task and If this issue was still open by then I will do it again and will apply your reviews.
Sorry again and thank you.

**Comment by FKSRipple** [comment]
@Soolan np, I'll copy the instructions that I see in GitHub to check out the branch, it seems like it's still stored somewhere:

```
git checkout -b Soolan-987-cleanup-combine develop
curl -L https://github.com/ripple/ripple-lib/pull/1014.patch | git am -3
```

**Comment by FKSRipple** [comment]
Closing this PR but feel free to resubmit using the `git checkout` instructions above to get back your old branch

**Pull request closed without merging**